### PR TITLE
Add client processor for CloudStack datacenter config

### PR DESCRIPTION
**Pull request by cxbrowne1207**
*Issue #, if available:*
https://github.com/aws/eks-anywhere-internal/issues/968
*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cluster/cloudstack_test.go [hunk 1]
```diff
@@ -15,3 +22,59 @@ func TestParseConfigMissingCloudstackDatacenter(t *testing.T) {
...
+	}
+
+	wantError := errors.New("test error")
+
+	client.EXPECT().Get(ctx, "datacenter", "default", &anywherev1.CloudStackDatacenterConfig{}).Return(wantError)
+
+	_, err := b.Build(ctx, client, cluster)
+	g.Expect(err).To(MatchError(ContainSubstring("building Config from a cluster client")))
+
+	client.EXPECT().Get(ctx, "datacenter", "default", &anywherev1.CloudStackDatacenterConfig{}).Return(nil).DoAndReturn(
```
Comment (g-gaston) [inline pkg/cluster/cloudstack_test.go:66]: can we use fake.Client for this? it makes the tests so much easier to work with

File: pkg/cluster/cloudstack_test.go [hunk 2]
```diff
@@ -15,3 +22,59 @@ func TestParseConfigMissingCloudstackDatacenter(t *testing.T) {
 	g.Expect(err).To(Not(HaveOccurred()))
 	g.Expect(got.CloudStackDatacenter).To(BeNil())
 }
+
+func TestDefaultConfigClientBuilderCloudStackCluster(t *testing.T) {
```
Comment (g-gaston) [inline pkg/cluster/cloudstack_test.go:26]:
```suggestion
func TestDefaultConfigClientBuilderBuildCloudStackCluster(t *testing.T) {
```

**Review by taneyland — APPROVED** [review]

File: pkg/cluster/cloudstack_test.go [hunk 3]
```diff
@@ -15,3 +20,64 @@ func TestParseConfigMissingCloudstackDatacenter(t *testing.T) {
...
+			APIVersion: anywherev1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "datacenter",
+			Namespace: "default",
+		},
+	}
+
+	client := fake.NewClientBuilder().WithObjects().Build()
+	_, err := b.Build(ctx, clientutil.NewKubeClient(client), cluster)
```
Comment (g-gaston) [inline pkg/cluster/cloudstack_test.go:41]: you can also do `test.NewFakeKubeClient()`

File: pkg/cluster/cloudstack_test.go [hunk 4]
```diff
@@ -15,3 +20,64 @@ func TestParseConfigMissingCloudstackDatacenter(t *testing.T) {
...
+	g.Expect(err).To(MatchError(ContainSubstring("building Config from a cluster client")))
+
+	client = fake.NewClientBuilder().WithObjects(datacenter).Build()
+	config, err := b.Build(ctx, clientutil.NewKubeClient(client), cluster)
```
Comment (g-gaston) [inline pkg/cluster/cloudstack_test.go:45]: could we separate this into two tests?
or at least use `t.Run`

File: pkg/cluster/cloudstack_test.go [hunk 5]
```diff
@@ -15,3 +20,64 @@ func TestParseConfigMissingCloudstackDatacenter(t *testing.T) {
...
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "datacenter",
+			Namespace: "default",
+		},
+	}
+
+	client := fake.NewClientBuilder().WithObjects().Build()
+	_, err := b.Build(ctx, clientutil.NewKubeClient(client), cluster)
+	g.Expect(err).To(MatchError(ContainSubstring("building Config from a cluster client")))
```
Comment (g-gaston) [inline pkg/cluster/cloudstack_test.go:42]: isn't this testing the same as `TestDefaultConfigClientBuilderBuildCloudStackClusterFailure`

Comment (cxbrowne1207) [reply pkg/cluster/cloudstack_test.go:42]: Ah, apologize. This was suppose to be removed. Thanks for catching that!

**Review by g-gaston — APPROVED** [review]

**Pull request merged**